docs/proposed/http-introducer-internals.rst

```diff
@@ -0,0 +1,192 @@
+
+The result is an unguessable self-authenticating URL which can be used to establish a confidential channel to the HTTP Introducer.
+
+Protocol
+--------
+
+The long-running process operates a TLS server on its configured endpoint.
+It uses the configured certificate and private key for the necessary TLS negotiation.
+On top of TLS,
+it runs a WebSocket server with two endpoints.
+
+The server does not require the client to present a certificate.
+The client requires the server to present a certificate with an SPKI hash matching that in the *introducer fURL*.
+
+.. TODO: Add docs about the WebSocket protocol negotiation that happens for the pub/sub protocol
+.. TODO: Collapse the two simplex endpoints into one duplex endpoint.
+   If client sends server a message, it's publishing an announcement.
+   If server sends client a message, it's delivering an announcement someone published.
+.. TODO: Add discussion of connection management, esp reconnection on lost connection.
+
+/<swissnum>/publish
+~~~~~~~~~~~~~~~~~~~
+
+This endpoint accepts JSON messages that contain storage announcements.
+A storage announcement is a small JSON document that looks something like this::
+
+   {"v0-p46y...":
+     { "ann":
+       { anonymous-storage-FURL": "pb://sokl...@192.168.69.247:44801/eqpw..."
+       , "nickname": "storage001"
+       }
+     }
+   }
+
+The top-level key is a v0 node public key identifying the node the announcement describes.
+The next level key is always the string ``"ann"``.
+Keys in the next level give specific information for how to connect to the storage server.
+Depending on how the storage server is exposed,
+these details may vary in structure.
+
+A storage announcement is created by a storage server trying to expose a storage service.
+The storage server can send one of these messages at any time to initially announce itself or update a previous announcement.
+
+/<swissnum>/subscribe
+~~~~~~~~~~~~~~~~~~~~~
+
+This endpoint produces JSON messages that contain storage announcements.
+On initial connection every storage announcement that has been received by the server is produced.
+Subsequently,
+whenever a storage announcement is published it is delivered to all subscribers.
+
+Failure Modes
+-------------
+
+HTTP Introducer imposes the following requirements:
+
+* It most be possible to establish a network connection from storage servers and client nodes to HTTP introducer.
+  * If these connections cannot be established then announcements cannot be published or delivered to subscribers.
+* Storage servers must retain the *introducer fURL*.
+  If they lose it they can no longer publish announcements.
+* Clients must retain the *introducer fURL*.
+  If they lose it they can no longer subscribe to announcements.
+* An administrator must retain the HTTP Introducer state.
+  It must be kept secret or another agent will be able to pose as the introducer
+  (however all they can do is deny service).
```

Comment:
Yea they can do that but they can also do that without impersonating the introducer - or at least, doing that only requires knowing the introduction NURL (name introduced by #896 ) since anyone can announce anything they like.

I think I wrote this with the grand future of public introducer furls in mind (with zkaps and grid manager to alleviate all access control concerns).  I'll try to tweak it to reflect both this and other possible realities.